packages/create-cloudflare/src/helpers/args.ts

```diff
@@ -1,88 +1,239 @@
-		})
-		.option("open", {
+		},
+		{
+			name: "open",
 			type: "boolean",
 			default: true,
 			description:
 				"Opens the deployed application in your browser (this option is ignored if the application is not deployed)",
-		})
-		.option("existing-script", {
+		},
+		{
+			name: "existing-script",
+			description: `The name of an existing Cloudflare Workers script to clone locally.
+
+        When "--existing-script" is specified, "deploy" will be ignored.
+        `,
 			type: "string",
 			requiresArg: true,
 			hidden: true,
-		})
-		.option("template", {
+		},
+		{
+			name: "template",
 			type: "string",
 			requiresArg: true,
-			description:
-				"A degit compatible string or the url to a git repository (with optionally a directory path) containing the C3 template to use",
-		})
-		.option("accept-defaults", {
+			description: `An external template to be used when creating your project.
+
+        Any "degit" compatible string may be specified. For example:
+
+        npm create cloudflare my-project -- --template github:user/repo
+        npm create cloudflare my-project -- --template git@github.com:user/repo
+        npm create cloudflare my-project -- --template https://github.com/user/repo
+        npm create cloudflare my-project -- --template git@github.com:user/repo#dev (branch)
+        npm create cloudflare my-project -- --template git@github.com:user/repo#v1.2.3 (tag)
+        npm create cloudflare my-project -- --template git@github.com:user/repo#1234abcd (commit)
+
+        Note that subdirectories may also be used. For example:
+
+        npm create cloudflare -- --template https://github.com/cloudflare/workers-sdk/templates/worker-r2
+        `,
+		},
+		{
+			name: "accept-defaults",
 			alias: "y",
 			type: "boolean",
 			description:
 				"Use all the default C3 options (each can also be overridden by specifying it)",
-		})
-		.option("auto-update", {
+		},
+		{
+			name: "auto-update",
 			type: "boolean",
 			default: C3_DEFAULTS.autoUpdate,
 			description: "Automatically uses the latest version of C3",
-		})
-		.option("wrangler-defaults", { type: "boolean", hidden: true })
+		},
+		{
+			name: "wrangler-defaults",
+			description: "Use special defaults for `wrangler init`",
+			type: "boolean",
+			hidden: true,
+		},
+		{
+			name: "help",
+			description: "Show help and exit",
+			type: "boolean",
+			hidden: true,
+		},
+	],
+};
+
+export const parseArgs = async (argv: string[]): Promise<Partial<C3Args>> => {
+	const doubleDashesIdx = argv.indexOf("--");
+	const c3Args = argv.slice(
+		0,
+		doubleDashesIdx < 0 ? undefined : doubleDashesIdx
+	);
+	const additionalArgs =
+		doubleDashesIdx < 0 ? [] : argv.slice(doubleDashesIdx + 1);
+
+	const yargsObj = yargs(hideBin(c3Args))
+		.scriptName("create-cloudflare")
+		.usage("$0 [args]")
 		.version(version)
 		.alias("v", "version")
+		.help(false)
+		.positional("potato", { type: "string" })
```

Comment:
if add a `potato` arg I think we should also add a `carrot` one, either that or we should remove this argument 🥔